lib/genesis/helpers.go

```diff
@@ -232,33 +336,249 @@ func buildRawArrayInterface(a []interface{}, kv *keyValue) error {
+		encode, err = scale.Marshal(t)
+		if err != nil {
+			return nil, err
+		}
+	case []interface{}:
+		enc := make([][32]byte, len(t))
+		var accAddr accountAddr
+		for index, add := range t {
+			copy(accAddr[:], crypto.PublicAddressToByteArray(common.Address(add.(string))))
+			enc[index] = accAddr
+		}
+		encode, err = scale.Marshal(enc)
+		if err != nil {
+			return nil, err
+		}
+	case []string:
+		for _, add := range t {
+			accID := crypto.PublicAddressToByteArray(common.Address(add))
+			encode = append(encode, accID...)
+		}
+		encode, err = scale.Marshal(encode)
+		if err != nil {
+			return nil, err
+		}
+	case string:
+		if t == "NotForcing" { //TODO: check if this is ok
+			encode, err = scale.Marshal(uint8(0))
+			if err != nil {
+				return nil, err
+			}
+		}
+	case [][]interface{}:
+		for _, data := range t {
+			for _, v := range data {
+				var accAddr accountAddr
+				switch v1 := v.(type) {
+				case string:
+					copy(accAddr[:], crypto.PublicAddressToByteArray(common.Address(v1)))
+					encode = append(encode, accAddr[:]...)
+				case float64:
+					var bytesVal []byte
+					bytesVal, err = scale.Marshal(big.NewInt(int64(v1)))
+					if err != nil {
+						return nil, err
+					}
+					encode = append(encode, bytesVal...)
+				}
+			}
+		}
+
+		encode, err = scale.Marshal(encode)
+		if err != nil {
+			return nil, err
+		}
+	default:
+		return nil, fmt.Errorf("errror")
+	}
+	return encode, nil
+}
+
+func generateContractKeyValue(c *contracts, prefixKey string, res map[string]string) error {
+	var (
+		key string
+		err error
+	)
+	// First field of contract is the storage key
+	val := reflect.ValueOf(c)
+	if k := reflect.Indirect(val).Type().Field(0).Name; k == "CurrentSchedule" {
+		key, err = generateStorageKey(prefixKey, k)
+		if err != nil {
+			return err
+		}
+	}
+
+	encode, err := scale.Marshal(c)
+	if err != nil {
+		return err
+	}
+	res[key] = common.BytesToHex(encode)
+	return nil
+}
+
+func generateKeyValue(s interface{}, prefixKey string, res map[string]string) error {
+	val := reflect.ValueOf(s)
+	n := reflect.Indirect(val).NumField()
+	for i := 0; i < n; i++ {
+		val := reflect.ValueOf(s)
+		storageKey := reflect.Indirect(val).Type().Field(i).Name
+		if storageKey == "Phantom" { //TODO: figure out what to do with Phantom as its value is null
+			continue
+		}
+		key, err := generateStorageKey(prefixKey, storageKey)
+		if err != nil {
+			return err
+		}
+		value, err := generateStorageValue(s, i)
+		if err != nil {
+			return err
+		}
+		res[key] = common.BytesToHex(value)
+	}
+	return nil
+}
+
 func formatKey(kv *keyValue) (string, error) {
 	switch {
-	case reflect.DeepEqual([]string{"grandpa", "authorities"}, kv.key):
+	case reflect.DeepEqual([]string{"Grandpa", "authorities"}, kv.key):
 		kb := []byte(`:grandpa_authorities`)
 		return common.BytesToHex(kb), nil
-	case reflect.DeepEqual([]string{"system", "code"}, kv.key):
+	case reflect.DeepEqual([]string{"System", "changesTrieConfig", "code"}, kv.key):
```

Comment:
I'm a bit confused on the addition of `changesTrieConfig` here, is it basically just another potential field under `System`?

Reply:
Currently, we are not using `changesTrieConfig `. I have removed it and added a todo.